src/node/browser.ts

```diff
@@ -275,23 +291,45 @@ $doc.body.appendChild($style);
     return { ...this.currentStory, buffer, elapsedTime };
   }
 
-  async setCurrentStory(kind: string, story: string, count: number ) {
-    this.currentStory = { kind, story, count };
-    const data = {
-      key: "storybook-channel",
-      event: {
-        type: "setCurrentStory",
-        args: [
-          {
-            kind,
-            story,
-          },
-        ],
-        from: "zisui",
-      }
-    };
-    this.debug("Set story", kind, story);
+  async setCurrentStory(s: Story & { count: number }) {
+    this.currentStory = s;
+    this.debug("Set story", s.kind, s.story);
+    const data = this.createPostmessageData(s);
     await this.page.evaluate((d: typeof data) => window.postMessage(JSON.stringify(d), "*"), data);
   }
 
+  private createPostmessageData(story: Story) {
+    // REMARKS
+    // zisue uses storybook post message channel, which is Storybook internal API.
+    switch (story.version) {
+      case "v4" :
+        return {
+          key: "storybook-channel",
+          event: {
+            type: "setCurrentStory",
+            args: [
+              {
+                kind: story.kind,
+                story: story.story,
+              },
+            ],
+            from: "zisui",
+          }
+        };
+      case "v5" :
+        return {
+          key: "storybook-channel",
+          event: {
+            type: "setCurrentStory",
+            args: [
+              {
+                storyId: story.id,
```

Comment:
With storybook v5, we send `storyId` to storybook channel rather than `kind` and `story` .